### ISSUE TITLE
Switch to p7zip_jll for unpacking

### ISSUE THREAD

**Issue by iamed2** [oxinabox/DataDeps.jl#116]
p7zip_jll supports all platforms and archive formats so it would probably reduce headaches. It has automatic format recognition as well. It would reduce `unpack_cmd` to a single function with a single condition (tar pipeline or not).

**Comment by oxinabox** [comment]
I am of two minds.

**Cons:** keeping light of dependencies is good.
Particularly when those dependencies are binary that are hard to support pre-julia 1.3 (since need to also add BinaryProvider as a dependency.)
We want to support 1.0 because that is one (of many) differences between DataDeps and the Pkg Artifacts stuff.

**Pros:** On the other-hand I have seen this break so many times.
I don't get why people don't include `unzip` in their standard linux install / docker image.
Why do people not do that, It is a tiny tiny thing to add?
Further this would give us support for .7z on all platforms, which we don't have right now.

**Comment by iamed2** [comment]
DataDeps already depends on BinaryBuilder pre-1.3 via HTTP via MbedTLS.

**Comment by oxinabox** [comment]
fair enough. Lets do this.

**Comment by juliohm** [comment]
Is there any update regarding this issue? We bumped into it just now. Manjaro doesn't have unzip installed by default, and the unpack stage fails.

cc @zerefwayne

**Comment by oxinabox** [comment]
There is no update on this.

**Comment by juliohm** [comment]
@zerefwayne do you think this issue is something you would like to spend time on? We could try to submit a PR to DataDeps.jl or try to fix things on our side with packages such as ZipFile.jl

**Comment by zerefwayne** [comment]
Yes, I can look into it. @oxinabox Instead of adding one more dependency, can we check for `unzip` and install it if not already present?

**Comment by juliohm** [comment]
@zerefwayne you mean an optional dependency? I think it would be overkill. In my opinion most users of DataDeps.jl would be fine with a Zip utility installation as a dependency.

If it is not fine, then maybe consider using Requires.jl to add optional dependencies. Again, I think it is overkill.

**Comment by zerefwayne** [comment]
I mean we don't have to install anything new. We can simply check if unzip exists or not. Then we can install unzip using bash commands and proceed with normal flow.

**Comment by juliohm** [comment]
Please keep in mind that we *always* want to be cross-platform so bash commands are out of the game. The whole point of using a package is that the package already took care of installing things correctly across different OSes.

**Comment by zerefwayne** [comment]
ZipFile.jl is a good solution but they are missing a unzip function

```julia
r = ZipFile.Reader("/tmp/example.zip");
for f in r.files
    println("Filename: \$(f.name)")
    write(stdout, read(f, String));
end
```

**Comment by oxinabox** [comment]
`p7zip_jll` is a better solution. It works for all file types.
Not just `.zip`.

**Comment by zerefwayne** [comment]
@oxinabox I had a look at p7zip_dll and it's working fine. Since it's cross-platform, do we need the OS checks?

```julia
if Sys.isunix() && Sys.KERNEL != :FreeBSD

if Sys.KERNEL == :FreeBSD

if Sys.iswindows()
```

The extensions that we are supporting are supported by 7z and it's cross-platform. So I think we can really simplify the unpack command to just 5-6 lines.

**Comment by oxinabox** [comment]
> The extensions that we are supporting are supported by 7z and it's cross-platform. So I think we can really simplify the unpack command to just 5-6 lines.

Correct.

**Comment by omus** [comment]
Fixed in #127

**Comment by omus** [comment]
Fix is included in DataDeps 0.7.5
